enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,398 @@
+  used by the cluster
+
+
+## Existing Configuration Interfaces
+
+### Summary
+
+First, here’s a summary of how an administrator may configure host networking
+in a cluster.  More details about each interface are in the following sections.
+
+#### First Boot
+
+When an RHCOS host first boots, the initramfs environment must be able to reach
+the Machine Config Server for Ignition to succeed.  By default, dracut will
+attempt to do DHCPv4 and DHCPv6 on all interfaces and wait for one of them to
+work.  To change this behavior, an admin can:
+
+* Change the `ip=<...>` kernel argument(s) via PXE configuration or at the grub
+  command line during boot
+* If at least one network has DHCP to allow dracut to succeed, custom ignition
+  additions can configure additional networking
+* If using the RHCOS live ISO/PXE ramdisk environment with the
+  coreos-installer, use the `nmtui` Network Manager text UI to configure
+  networking before proceeding
+
+#### Further Network Customization
+
+Beyond the minimum needed to bootstrap the cluster, further host network
+configuration customizations can be applied in the following ways:
+
+* Customize the ignition configuration at install time to include more
+  NetworkManager configuration files
+* Create `MachineConfig` resources with NetworkManager configuration files and
+  have them applied by the installer at install time, or apply them to the
+  cluster via the API any time post-install
+
+### Dracut in initfamfs
+
+In the initramfs environment at boot time,
+[dracut](https://github.com/dracutdevs/dracut) is the first tool that sets up
+networking on the host.
+
+Configuring dracut is done via [kernel command line
+arguments](https://mirrors.edge.kernel.org/pub/linux/utils/boot/dracut/dracut.html#_network).
+This can be customized via PXE configuration or at the grub command line during
+boot.
+
+Networking at this stage of the installation process is important, as the
+installer generates a “pointer ignition config”, or a small ignition stub that
+just points to a URL for where to download the full ignition configuration from
+the Machine Config Server.  In other words, when a new RHCOS host boots for the
+first time, dracut must configure networking just enough for ignition to be
+able to reach the Machine Config Server.
+
+A [past bug against
+dracut](https://bugzilla.redhat.com/show_bug.cgi?id=1787620) ensured that there
+was a way to get dracut to attempt both DHCPv4 (IPv4) and DHCPv6 (IPv6) and not
+fail if one of them worked.  We also [changed
+OpenShift](https://bugzilla.redhat.com/show_bug.cgi?id=1793591) to use the
+`ip=dhcp,dhcp6` configuration by default in RHCOS.  This at least got default
+behavior far enough along that it will work in either an IPv4 or IPv6
+environment.
+
+It does not help if one of the host’s network interfaces does not provide
+connectivity to the Machine Config Server (for example, if a VLAN on an
```

Comment:
Is that saying "Running DHCP on interfaces that can't reach the MCS doesn't help us reach the MCS"? Um, sure...

Reply:
yeah, kind of awkward.  I reworded this paragraph.  Hopefully it's better